components/turnToAngle.py

```diff
@@ -1,101 +1,97 @@
-            
-            
-            self.change = self.nextHeading - self.heading
-            if self.change > 180:
-                self.change -= 360
-            elif self.change < -180:
-                self.change += 360
-            
-            if abs(self.change) > 90:
-                self.speed = .25
-            elif abs(self.change) <= 90 and abs(self.change) > 20:
-                self.speed = .2
-            elif abs(self.change) <= 20:
-                self.speed = .15
-
-            if self.setSpeed == True:
-                if self.change > 0:
-                    self.driveTrain.setTank(-1 * self.speed, self.speed)
-                else:
-                    self.driveTrain.setTank(self.speed, -1 * self.speed)
-            
-            if (self.heading <= self.nextHeading + self.tolerance and self.heading >= self.nextHeading - self.tolerance):
-                self.setSpeed = False
-                self.nextOutput = self.PIDController.calculate(measurement = self.heading, setpoint = self.nextHeading)
-                self.driveTrain.setTank(-1 * self.nextOutput, self.nextOutput)
+    def setAngle(self, angle):
+        """Sets the desired turn angle"""
+        self.turnAngle = angle
 
-    @feedback
-    def outputDisplay(self):
-        return self.nextOutput
+    def start(self):
+        self.starting = True
 
-    @feedback
-    def nextHeadingDisplay(self):
-        return self.nextHeading
+    @state(first = True)
+    def idling(self):
+        """Stays in this state until started"""
+        if self.turnAngle != 0:
+            self.next_state("turn")
+        else:
+            log.error("Must have an angle to turn to")
+            self.next_state("idling")
+
+
+    def calcHeading(self):
+        """Calculates how far away from the desired angle the bot is"""
+        self.running = True
+        self.starting = False
+        self.nextHeading = self.initialHeading + self.turnAngle
 
-    @feedback
-    def setSpeedDisplay(self):
-        return self.setSpeed
-    
-    def stop(self):
-        self.nextOutput = 0
-        self.PIDController.reset()
-        
         if self.nextHeading > 360:
             self.nextHeading -= 360
-        
-        self.isRunning = False
-        self.initialHeading = self.heading
-        self.setSpeed = True
+        elif self.nextHeading < 0:
+            self.nextHeading += 360
 
+        self.change = self.nextHeading - self.navx.getFusedHeading()
+        if self.change > 180:
+            self.change -= 360
+        elif self.change < -180:
+            self.change += 360
 
-    def execute(self):
-        self.output()
-        self.PIDController = controller.PIDController(Kp= self.P, Ki= self.I, Kd= self.D, period = self.time)
-        self.heading = self.navx.getFusedHeading()
+    def setSpeedFunc(self):
+        """Determines the speed based off of the distance from the desired angle"""
+        self.calcHeading()
+        if abs(self.change) > 90:
```

Comment:
Could simplify from 3 if elsif to a if/elseif/else or to 2 if() that change from a default (speed = speed for > 90 if(90 <) (set < 90 speed); if(<slow) set slow;